### PR TITLE
feat(gcloud): honor `CLOUDSDK_CORE_PROJECT` env variable

### LINKED ISSUE #2540: gcloud module does not reflect CLOUDSDK_CORE_PROJECT
#### Current Behavior
The `gcloud` module only reads the current project as it's shown in the files `~/.config/gcloud/active_config` and `~/.config/gcloud/configurations/config_{CONFIG NAME}`.

#### Expected Behavior
Starship displays the value of the variable `CLOUDSDK_CORE_PROJECT` if it's set, as this overrides the active configuration.

#### Additional context/Screenshots
```
[2021-04-01 21:20:33]» cat ~/.config/gcloud/configurations/config_thames
[core]
account = river@thames.co.uk
project = foo-bar

[compute]
zone = us-central1-f
region = us-central1

[run]
platform = managed
region = us-east1


[2021-04-01 20:53:05]❮ gcloud config list                                                   
[compute]                                                                                   
region = us-central1                                                    
zone = us-central1-f                                                                        
[core]                                                                                      
account = river@thames.co.uk                                                        
disable_usage_reporting = True                                              
project = foo-bar
[run]                               
platform = managed                  
region = us-east1               

Your active configuration is: [thames]

[2021-04-01 20:43:05]» export CLOUDSDK_CORE_PROJECT=baz-qux

[2021-04-01 20:46:05]» gcloud config set project foo-bar
WARNING: Property [project] is overridden by environment setting [CLOUDSDK_CORE_PROJECT=baz-qux]
Updated property [core/project].                                                            
                                                         
[2021-04-01 20:53:05]❮ gcloud config list                                                   
[compute]                                                                                   
region = us-central1                                                    
zone = us-central1-f                                                                        
[core]                                                                                      
account = river@thames.co.uk                                                        
disable_usage_reporting = True                                              
project = baz-qux
[run]                               
platform = managed                  
region = us-east1               

Your active configuration is: [thames]

[infrastructure/data-pipeline][origin ledp][  casey@learningequality.org (scratch-casey)]
[2021-04-01 21:02:17]» echo $CLOUDSDK_ACTIVE_CONFIG_NAME

<No output shown>
```

Official documentation: https://cloud.google.com/sdk/docs/configurations#setting_configuration_properties

#### Possible Solution
Check if `CLOUDSDK_CORE_PROJECT` is set in the shell and read its value if it is, using that as the string shown in the starship command prompt.

#### Environment
- Starship version: 0.50.0
- zsh version: 5.4.2
- Operating system: Linux Mint 19.3
- Terminal emulator: gnome-terminal
- Git Commit Hash: 0558b38e
- Branch/Tag: v0.50.0
- Rust Version: rustc 1.49.0 (e1884a8e3 2020-12-29)
- Rust channel: stable-x86_64-unknown-linux-gnu (default) release
- Build Time: 2021-02-07 07:25:01

#### Starship Configuration

```
[gcloud]
disabled = false
style = "blue"
symbol = " "
format = '[\[${symbol}${account}( \($project\))\]]($style)'
```

### PR DESCRIPTION
**Pull request by vladimyr**
#### Description

Honor `CLOUDSDK_CORE_PROJECT` environment variable.

#### Motivation and Context

This is essentially #2588 rebased/stacked onto #2592
Closes #2588
Closes #2540

#### How Has This Been Tested?

- [ ] I have tested using **MacOS**
- [x] I have tested using **Linux**
- [ ] I have tested using **Windows**

#### Checklist:

- [ ] I have updated the documentation accordingly.
- [x] I have updated the tests accordingly.

---

/cc @cymruu

### CODE REVIEW

**Comment by andytom** [comment]
@vladimyr, this LGTM can you fix the merge conflicts and we should be good to go here.

**Review by andytom — APPROVED** [review]

File: src/modules/gcloud.rs [hunk 1]
```diff
@@ -302,6 +305,37 @@ project = abc
...
+
+        let actual = ModuleRenderer::new("gcloud")
+            .env("CLOUDSDK_CORE_PROJECT", "env_project")
+            .env("CLOUDSDK_CONFIG", dir.path().to_string_lossy())
+            .config(toml::toml! {
+                [gcloud]
+                format = "on [$symbol$project]($style) "
+            })
+            .collect();
+        let expected = Some(format!("on {} ", Color::Blue.bold().paint("☁️ env_project")));
```
Comment (andytom) [inline src/modules/gcloud.rs:333]: Looks like I broke this when I tried to fix the merge conflicts, I think this should fix it 
```suggestion
        let expected = Some(format!("on {} ", Color::Blue.bold().paint("☁️  env_project")));
```

Comment (vladimyr) [reply src/modules/gcloud.rs:333]: I'll take a look once I get back to my machine. ⌛

Comment (vladimyr) [reply src/modules/gcloud.rs]: Yup, that was it!

**Review by andytom — APPROVED** [review]

**Comment by andytom** [comment]
Thank you @vladimyr and @cymruu for your contribution.

**Pull request merged**